gateleen-runconfig/src/main/java/org/swisspush/gateleen/runconfig/RunConfig.java

```diff
@@ -597,11 +608,25 @@ private void handleRequest(final RoutingContext ctx) {
                     return;
                 }
                 if (PackingHandler.isPacked(request)) {
-                    request.bodyHandler(new PackingHandler(request, new QueuingHandler(vertx, redisProvider, request, monitoringHandler)));
+                    request.bodyHandler(new PackingHandler(
+                            request,
+                            new QueuingHandler(
+                                    vertx,
+                                    redisProvider,
+                                    request,
+                                    monitoringHandler,
+                                    queueSplitter != null ? queueSplitter : new NoOpQueueSplitter()
+                            )
+                    ));
                 } else {
                     if (QueuingHandler.isQueued(request)) {
                         setISO8601Timestamps(request);
-                        request.bodyHandler(new QueuingHandler(vertx, redisProvider, request, monitoringHandler));
+                        request.bodyHandler(new QueuingHandler(
+                                vertx,
+                                redisProvider,
+                                request,
+                                monitoringHandler,
+                                queueSplitter != null ? queueSplitter : new NoOpQueueSplitter()));
```

Comment:
You already do this logic in the QueuingHandler constructor. I would suggest to just use the `queueSplitter` even if null and do this check in the Constructor. When `null` is passed you could create a `NoOpQueueSplitter`